crates/chain/tests/test_keychain_txout_index.rs

```diff
@@ -44,29 +52,116 @@ fn spk_at_index(descriptor: &Descriptor<DescriptorPublicKey>, index: u32) -> Scr
         .script_pubkey()
 }
 
+#[test]
+fn append_keychain_derivation_indices() {
+    let mut lhs_di = BTreeMap::<[u8; 8], u32>::default();
+    let mut rhs_di = BTreeMap::<[u8; 8], u32>::default();
+    lhs_di.insert(descriptor_ids(0), 7);
+    lhs_di.insert(descriptor_ids(1), 0);
+    rhs_di.insert(descriptor_ids(1), 3);
+    rhs_di.insert(descriptor_ids(1), 5);
+    lhs_di.insert(descriptor_ids(2), 3);
+    rhs_di.insert(descriptor_ids(3), 4);
```

Comment:
To match the below comments on the expected results this should be as below, also can removed the unneeded insert 3 for rhs_di for descriptor 1.
```suggestion
    lhs_di.insert(descriptor_ids(0), 7);
    lhs_di.insert(descriptor_ids(1), 0);
    lhs_di.insert(descriptor_ids(2), 3);
    
    rhs_di.insert(descriptor_ids(0), 3); // value less than lhs 0
    rhs_di.insert(descriptor_ids(1), 5); // value more than lhs 1
    rhs_di.insert(descriptor_ids(3), 4); // key doesn't exist in lhs
```